File: src/main/java/com/EmosewaPixel/pixellib/recipes/SimpleRecipeBuilder.java
package com.EmosewaPixel.pixellib.recipes;

//This is a builder for Simple Machine Recipes
public class SimpleRecipeBuilder extends AbstractRecipeBuilder<SimpleMachineRecipe, SimpleRecipeBuilder> {
    public SimpleRecipeBuilder(AbstractRecipeList<SimpleMachineRecipe, SimpleRecipeBuilder> list) {
        super(list);
    }

    @Override
    public SimpleMachineRecipe build() {
        if (getInputs().size() <= getRecipeList().getMaxInputs() && getOutputs().size() <= getRecipeList().getMaxOutputs())
            return new SimpleMachineRecipe(getInputs().toArray(), getConsumeChances().toArray(new Float[getConsumeChances().size()]), getOutputs().toArray(), getOutputChances().toArray(new Float[getOutputChances().size()]), getTime());
        return SimpleMachineRecipe.EMPTY;
    }
}

File: src/main/java/com/emosewapixel/pixellib/recipes/SimpleRecipeBuilder.kt
package com.emosewapixel.pixellib.recipes

//This is a builder for Simple Machine Recipes
class SimpleRecipeBuilder(list: AbstractRecipeList<SimpleMachineRecipe, SimpleRecipeBuilder>) : AbstractRecipeBuilder<SimpleMachineRecipe, SimpleRecipeBuilder>(list) {
    override fun build() = SimpleMachineRecipe(inputs.map { it.first }.toTypedArray(), inputs.map { it.second }.toTypedArray(), fluidInputs.toTypedArray(), outputs.map { it.first }.toTypedArray(), outputs.map { it.second }.toTypedArray(), fluidOutputs.toTypedArray(), time)
}